From 478d3771ea6ba7729f341377a1cca377e6d16235 Mon Sep 17 00:00:00 2001
From: Jianxun Zhang <jianxun.zhang@intel.com>
Date: Sun, 5 Apr 2015 09:00:17 -0700
Subject: [PATCH] pstore: fix warnings in header file

Fix warnings when compiling without pstore enabled

In file included from arch/powerpc/platforms/pseries/nvram.c:21:0:
   include/linux/pstore.h:116:8: error: return type defaults to 'int' [-Werror=return-type]
    static pstore_register_extra_dumper(struct pstore_extra_dumper *extra_dumper)
           ^
   include/linux/pstore.h:120:8: error: return type defaults to 'int' [-Werror=return-type]
    static pstore_unregister_extra_dumper(struct pstore_extra_dumper *extra_dumper)
           ^
include/linux/pstore.h:116:8: error: 'pstore_register_extra_dumper' defined but not used [-Werror=unused-function]
    static pstore_register_extra_dumper(struct pstore_extra_dumper *extra_dumper)
           ^
include/linux/pstore.h:120:8: error: 'pstore_unregister_extra_dumper' defined but not used [-Werror=unused-function]
    static pstore_unregister_extra_dumper(struct pstore_extra_dumper *extra_dumper)

Change-Id: I3f777b83ffd38a3e28e0823ea12d5b1e7192217e
Tracked-On: https://android.intel.com/351346
Signed-off-by: Jianxun Zhang <jianxun.zhang@intel.com>
---
 include/linux/pstore.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/include/linux/pstore.h b/include/linux/pstore.h
index eec1d02261a0..97d3ae3c190d 100644
--- a/include/linux/pstore.h
+++ b/include/linux/pstore.h
@@ -113,11 +113,13 @@ pstore_cannot_block_path(enum kmsg_dump_reason reason)
 {
 	return false;
 }
-static pstore_register_extra_dumper(struct pstore_extra_dumper *extra_dumper)
+static int __maybe_unused
+pstore_register_extra_dumper(struct pstore_extra_dumper *extra_dumper)
 {
 	return 0;
 }
-static pstore_unregister_extra_dumper(struct pstore_extra_dumper *extra_dumper)
+static int __maybe_unused
+pstore_unregister_extra_dumper(struct pstore_extra_dumper *extra_dumper)
 {
 	return 0;
 }
-- 
2.1.0

